File: src/models/news-question.ts
import { NewsQuestionChoice } from "./news-question-choice";
import { NewsQuestionKind } from "./news-question-kind";
import { Attachment } from "./attachment";

export type NewsQuestion = Readonly<{
  id: string;
  position: number;
  kind: NewsQuestionKind;

  fullTitle: string;
  title: string;

  shouldRespectMaximumChoices: boolean;
  maximumChoices: number;
  maximumLength: number;

  attachments: Array<Attachment>;
  content: string;

  answerID: string;
  shouldAnswer: boolean;

  choices: Array<NewsQuestionChoice>;
}> & {
  selectedAnswers?: Array<number>;
  textInputAnswer?: string;
  answered: boolean;
  answerDate?: Date;
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/NewsQuestion.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class NewsQuestion(
    val id: String,
    val position: Int,
    val kind: NewsQuestionKind,
    val fullTitle: String,
    val title: String,
    val shouldRespectMaximumChoices: Boolean,
    val maximumChoices: Int,
    val maximumLength: Int,
    val attachments: List<Attachment>,
    val content: String,
    val answerId: String,
    val shouldAnswer: Boolean,
    val choices: List<NewsQuestionChoice>,
    var selectedAnswers: List<Int>? = null,
    var textInputAnswer: String? = null,
    var answered: Boolean,
    var answerDate: LocalDateTime? = null
)
